packages/web3torrent/src/components/torrent-info/channels-list/ChannelsList.tsx

```diff
@@ -23,53 +24,62 @@ class ChannelsList extends React.Component<UploadInfoProps> {
   channelIdToTableRow(
     channelId: string,
     channels: Dictionary<ChannelState>,
-    wires: PaidStreamingWire[],
+    torrent: Torrent,
     participantType: 'payer' | 'beneficiary'
   ) {
     let channelButton;
     const channel = channels[channelId];
 
+    const wire = torrent.wires.find(
+      wire =>
+        wire.paidStreamingExtension.peerChannelId === channelId ||
+        wire.paidStreamingExtension.pseChannelId === channelId
+    );
     if (channel.status === 'closing') {
       channelButton = <button disabled>Closing ...</button>;
     } else if (channel.status === 'closed') {
       channelButton = <button disabled>Closed</button>;
+    } else if (channel.status === 'challenging') {
+      channelButton = <button disabled>Challenging</button>;
     } else {
-      channelButton = <button disabled>Running</button>;
+      channelButton = getPeerStatus(torrent, wire) ? (
+        <button disabled>Running</button>
+      ) : (
+        <button
+          className="button-alt"
+          onClick={_ => this.context.paymentChannelClient.challengeChannel(channelId)}
```

Comment:
this call results in a console error for now:
![image](https://user-images.githubusercontent.com/10502605/77346259-bc0d8e80-6d14-11ea-92fa-29ac4ad8caa8.png)
I'm assuming that's because it's not yet implemented.